### PR TITLE
Replace followProgress to avoid caching push/pull state

### PR DESCRIPTION
**Pull request by CameronDiver**
Change-type: patch
Signed-off-by: Cameron Diver <cameron@balena.io>

### CODE REVIEW

File: index.coffee [hunk 1]
```diff
@@ -19,16 +20,25 @@ tryExtractDigestHash = (evt) ->
...
-	return null
+awaitRegistryStream = (stream, onProgress) ->
+	contentHash = null
+	new Promise (resolve, reject) ->
+
+		jsonStream = JSONStream.parse()
+
+		jsonStream.on 'data', (evt) ->
+			if typeof evt isnt 'object'
+				onProgress({})
```
Comment (dfunckt) [inline index.coffee:31]: No need to call the handler in this case, in fact I think it might mess things up for some clients if it's an empty object.

**Review by dfunckt — APPROVED** [review]
Nice one 👍

File: index.coffee (hunk 1 above)
Comment (CameronDiver) [reply index.coffee:31]: I was keeping the iterface the same as `followProgress`, which does this: https://github.com/apocas/docker-modem/blob/master/lib/modem.js#L323

Comment (dfunckt) [reply index.coffee:31]: I see, that's fine then -- it seems clients already need to handle the empty object so we're good.

**Pull request merged**